### ISSUE TITLE
Compiler should warn for unused read-only properties

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#2976]
I've switched my usage of read-only fields nearly entirely to to read-only properties in C# 6, however, this has resulted in me no longer getting the benefit of the compiler warning when I fail to assign/use them.

For example, the following:

``` C#
    public class ClassWithReadOnlyFields
    {
        private readonly string _field1;
        private readonly string _field2 = "foo";
    }
```

results in:

```
CS0414	The field 'ClassWithReadOnlyField._field2' is assigned but its value is never used
CS0169	The field 'ClassWithReadOnlyField._field1' is never used
```

Whereas the following results in no warnings:

``` C#
    public class ClassWithReadOnlyPropertys
    {
        private string Property1
        {
            get;
        }

        private string Property2
        {
            get;
        } = "foo";
    }
```

It would be great to get warnings in this case, similar to fields.

**Comment by kindermannhubert** [comment]
Is this waiting for warning waves or? It's quite unfortunate there's no warning for this.
I see writing custom analyzer for this as waste of time - I suppose already present logic for detecting unused readonly fields in Roslyn could be used for detection of unused readonly auto-props with little changes. Am I wrong?

**Comment by CyrusNajmabadi** [comment]
> Is this waiting for warning waves or?

No, this hasn't been assigned and has not been felt to be important enough to warrant attention.

> I see writing custom analyzer for this as waste of time

Why?  It would be fairly simple and would get a solution out there far faster than waiting for Roslyn devs to get to it :)

> Am I wrong?

I think an easier way to do this would be to have an symbol-start-analyzer for named types, then create an IOperation analyzer for the members of that symbol.  Then just look for IPropertyReference operations.  At the end, report any private auto-props that were unused.

**Comment by CyrusNajmabadi** [comment]
Note: feel free to come to gitter.im/dotnet/roslyn or discord.gg/csharp (#roslyn channel) for help writing analyzers :)

**Comment by CyrusNajmabadi** [comment]
IDE0051 will be reported for these.
